lib/index.js

```diff
@@ -82,18 +86,24 @@ Optimizely.prototype.initialize = function() {
  */
 
 Optimizely.prototype.track = function(track) {
-  var payload = {};
+  // Use the new-style API (which is compatible with Classic and X)
+  var payload = {
+    type: 'event',
+    eventName: track.event(),
+    tags: track.properties()
+  };
   var revenue = track.revenue();
```

Comment:
Our spec asks that you should send a numeric value for `revenue` so let's not worry about it being non numeric. No you don't have to remove it if it `track.revenue()` doesn't return truthy value. It will return `undefined` if they did not send `properties.revenue` :)